### ISSUE TITLE
Multiple GPU usage

### ISSUE THREAD

**Issue by vswarup** [broadinstitute/CellBender#109]
Hi

I was wondering if it is possible to use multiple GPUs for the cellBender pipeline, currently there is no such option in the package. Please help

**Comment by sjfleming** [comment]
Hi @vswarup , 

Currently the code is written in such a way that it will not make use of more than one GPU.  Currently we do not have plans to enable the concurrent use of multi-GPUs.

Were you interested in decreasing the overall runtime?

**Comment by wzthhm** [comment]
Hi @sjfleming, I was interested in decreasing the overall runtime for a large number of samples. But when I ran cellbender on 4x GPUs, it always only use the first one. What should I do? Thanks

**Comment by sjfleming** [comment]
Yeah, @wzthhm , unfortunately, the code will not use more than one GPU at a time.  Currently, if you have 4 GPUs, the best thing to do would be to maybe run one separate cellbender run on one sample for each GPU.  If you can figure out a way to do that using multiple processes... I am not an expert there.

We developed CellBender to handle multiple samples by running it in parallel on the cloud, using a platform like Terra (app.terra.bio).  Instead of using a single machine with 4 GPUs to process a sample nearly 4x as fast, the idea would be to submit 4 jobs (one sample each) in parallel to run on separate 1 GPU machines.  If you have 4 GPUs locally, maybe you can do something similar?
